Backend/users/usersHelper.js

```diff
@@ -0,0 +1,51 @@
+require("dotenv").config();
+const jwt = require("jsonwebtoken");
+const { check, validationResult } = require("express-validator");
+
+const secret = process.env.JWT_SECRET || "default";
+
+const generateToken = (username, id) => {
+  const token = jwt.sign(
```

Comment:
I'm not sure about JWT, maybe it's good idea to make sessions instead? 
If someone gets hacked, it will be possible to use his account for 24h even if we ban him from the server.

Let's discuss that in private

Reply:
No, its not. JWT works through the signatures meaning it will be active until the time of token is expired

Reply:
More on this, while JWTs are designed to be stateless and do not inherently support, If we want to log out a malicious user, then we can add a flag in the database associated with that user, such as (isBlocked = false); by default, it will be false.
Why use this flag?
Using this flag we check when a user attempts to access protected routes. If the isBlocked flag is false and the session/token is not expired, then the user can access that page/route. However, if the isBlocked flag is true or the session/token has expired, then the user can't access any more pages. Does this make sense?
The crucial factor for blocking and logging out a user instantly is the flag, not the token or session. We block the user instantly based on the flag from the server side. I hope this clarifies things.
So the question is JWT or session storage? In my opinion, we should go with the JWT approach for scalability reasons and securities too. We can include any random string in the payload instead of user data to make it even more secure. Even if a user copies the token, they will only get a random string, which is not a bad deal. What do you think?

Reply:
Yes, You are right we use JWT here for Login(authentication) But not for checking block role of user, we would use session for that while retaining the scalability that comes with JWT and other reasons. However ,there is no problem I will implement local storage.

Reply:
What scalabilities do you see for JWT? In our case it's just the same as session, and that kills all benefits. You should not store your session token in the local storage either.